### PR TITLE
BLD: change SCIPY_USE_PYTHRAN default to `1`

### PR DESCRIPTION
**Pull request by rgommers**
This ensures we get more people to test with it, smoking out possible build issues.

Cc @serge-sans-paille 

EDIT: I checked that the `scipy-dev:latest` Dockerfile already has `pythran` installed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
             . venv/bin/activate
             pip install --install-option="--no-cython-compile" cython
             pip install numpy
-            pip install nose mpmath argparse Pillow codecov matplotlib "Sphinx!=3.1.0" asv
+            pip install nose mpmath argparse Pillow codecov matplotlib "Sphinx!=3.1.0" asv pythran
```
Comment (tylerjereddy) [inline .circleci/config.yml:52]: I went ahead and pushed in this missing install for CircleCI. Unrelated, but I wonder if that `nose` install can be removed at this point.

I'm less sure about the Appveyor failure from a very quick glance--the error output there is rather verbose and `pythran` is in the requirements file we have for appveyor.

**Comment by rgommers** [comment]
The Appveyor failure:

<details>

```
[00:05:12]   building 'scipy.optimize._group_columns' extension
[00:05:12] 
[00:05:12]   compiling C sources
[00:05:12]   C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -Ic:\python37-x64\lib\site-packages/pythran 
[00:05:12] -Ic:\python37-x64\lib\site-packages\numpy\core\include -Ic:\python37-x64\lib\site-packages\numpy\core\include -Ic:\python37-x64\include -Ic:\python37-x64\include -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE 
[00:05:12] -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IC:\Program Files 
[00:05:12] (x86)\Windows Kits\10\include\10.0.18362.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt -IC:\Program Files (x86)\Microsoft Visual 
[00:05:12] Studio 14.0\VC\INCLUDE -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um 
[00:05:12] -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt /EHsc 
[00:05:12] /Tpscipy/optimize/_group_columns.cpp /Fobuild\temp.win-amd64-3.7\Release\scipy/optimize/_group_columns.obj /std:c++14 /Zm1000
[00:05:12] 
[00:05:13]   Running from SciPy source directory.
[00:05:13] 
[00:05:13]   c:\python37-x64\lib\site-packages\numpy\distutils\system_info.py:838: UserWarning: Specified path c:\python37-x64\include\include is invalid.
[00:05:13] 
[00:05:13]     return self.get_paths(self.section, key)
[00:05:13]   _group_columns.cpp
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/types/variant_functor.hpp(115): warning C4812: obsolete declaration style: please use 
[00:05:13] '`anonymous-namespace'::pythonic::types::details::variant_functor_impl<Type>::~variant_functor_impl' instead
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/python/core.hpp(62): warning C4244: 'initializing': conversion from 'Py_ssize_t' to 'long', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/python/core.hpp(92): warning C4244: 'initializing': conversion from 'npy_intp' to 'long', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/python/core.hpp(98): warning C4244: '*=': conversion from 'const npy_intp' to 'long', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/python/core.hpp(144): warning C4244: 'initializing': conversion from 'Py_ssize_t' to 'long', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/types/int.hpp(170): warning C4244: 'return': conversion from 'long' to 'unsigned char', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/types/int.hpp(171): warning C4244: 'return': conversion from 'long' to 'signed char', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/types/int.hpp(172): warning C4244: 'return': conversion from 'long' to 'unsigned short', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/types/int.hpp(173): warning C4244: 'return': conversion from 'long' to 'short', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/builtins/id.hpp(28): warning C4244: 'return': conversion from 'uintptr_t' to 'long', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/builtins/id.hpp(33): warning C4244: 'return': conversion from 'uintptr_t' to 'long', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/builtins/id.hpp(38): warning C4244: 'return': conversion from 'uintptr_t' to 'long', possible loss of data
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/include/utils/shared_ref.hpp(101): warning C4521: '`anonymous-namespace'::pythonic::utils::shared_ref<std::vector<bool,std::allocator<_Ty>>>': multiple copy constructors 
[00:05:13] specified
[00:05:13]           with
[00:05:13]           [
[00:05:13]               _Ty=bool
[00:05:13]           ]
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/include/types/list.hpp(195): note: see reference to class template instantiation '`anonymous-namespace'::pythonic::utils::shared_ref<std::vector<bool,std::allocator<_Ty>>>' 
[00:05:13] being compiled
[00:05:13]           with
[00:05:13]           [
[00:05:13]               _Ty=bool
[00:05:13]           ]
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/include/types/list.hpp(513): note: see reference to class template instantiation '`anonymous-namespace'::pythonic::types::list<bool>' being compiled
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/include/types/numpy_gexpr.hpp(80): error C2955: '`anonymous-namespace'::pythonic::types::to_slice': use of class template requires template argument list
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/include/types/numpy_gexpr.hpp(40): note: see declaration of '`anonymous-namespace'::pythonic::types::to_slice'
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/include/types/numpy_gexpr.hpp(90): note: see reference to class template instantiation '`anonymous-namespace'::pythonic::types::extended_slice<C>' being compiled
[00:05:13]   c:\python37-x64\lib\site-packages/pythran\pythonic/include/types/numpy_gexpr.hpp(80): fatal error C1903: unable to recover from previous error(s); stopping compilation
[00:05:13]   Internal Compiler Error in C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe.  You will be prompted to send an error report to Microsoft later.
[00:05:13]   error: Command "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE 
[00:05:13] -Ic:\python37-x64\lib\site-packages/pythran -Ic:\py
[00:05:13] thon37-x64\lib\site-packages\numpy\core\include -Ic:\python37-x64\lib\site-packages\numpy\core\include -Ic:\python37-x64\include -Ic:\python37-x64\include -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE 
[00:05:13] -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um -IC:\Program Files 
[00:05:13] (x86)\Windows Kits\10\include\10.0.18362.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt -IC:\Program Files (x86)\Microsoft Visual 
[00:05:13] Studio 14.0\VC\INCLUDE -IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um 
[00:05:13] -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt /EHsc 
[00:05:13] /Tpscipy/optimize/_group_columns.cpp /Fobuild\temp.win-amd64-3.7\Release\scipy/optimize/_group_columns.obj /std:c++14 /Zm1000" failed with exit status 1
```

</details>

IIRC what is going on is that we need `clang-cl` and installed it on Azure (Windows builds pass there) but not on Appveyor. Honestly we should just get rid of Appveyor, it's completely duplicate. We wanted to wait a bit simply to be conservative, but at this point it's just a waste of CI resources and we're not testing anything that we don't also test on Azure. @tylerjereddy am I forgetting anything, or should we just stop using Appveyor?

**Comment by tylerjereddy** [comment]
Ok, I'm going to merge this because the only CI failure is Appveyor, which I've now disabled in a separate PR with removal of config files + I also purged the webhook for Appveyor from our repo (see reasons above).

Thanks Ralf

**Comment by rgommers** [comment]
> Ok, I'm going to merge this because the only CI failure is Appveyor, which I've now disabled in a separate PR with removal of config files + I also purged the webhook for Appveyor from our repo (see reasons above).

Great, thanks!

**Pull request merged**